tiatoolbox/models/engine/engine_abc.py

```diff
@@ -274,16 +269,41 @@ def _initialize_model_ioconfig(
 
         return model, None
 
-    def pre_process_patches(
+    @abstractmethod
+    def get_dataloader(
         self: EngineABC,
-        images: np.ndarray | list,
-        labels: list,
+        images: Path,
+        masks: Path | None = None,
+        labels: list | None = None,
+        ioconfig: ModelIOConfigABC | None = None,
     ) -> torch.utils.data.DataLoader:
         """Pre-process an image patch."""
         if labels:
             # if a labels is provided, then return with the prediction
             self.return_labels = bool(labels)
 
+        if isinstance(images, Path):
```

Comment:
This should be checking the mode. 
WSI can also be provided as numpy array similar to patches. The difference is that the size of patches is equal to size required by the network architecture.